provider crashes when importing cloudflare_ruleset

### Confirmation

- [X] My issue isn't already found on the issue tracker.
- [X] I have replicated my issue using the latest version of the provider and it is still present.

### Terraform and Cloudflare provider version

Terraform v1.4.6
Provider v3.35.0 to v4.7.1

### Affected resource(s)

cloudflare_ruleset

### Terraform configuration files

```hcl
terraform { 
  required_providers {
    cloudflare = {
      source = "cloudflare/cloudflare"
      version = "~> 4.0"
    }
  }
}

provider "cloudflare" {
  # token pulled from $CLOUDFLARE_API_TOKEN
}

resource "cloudflare_ruleset" "transform_rules" {
  kind    = "zone"
  name    = "transform rule to add a request header"
  phase   = "http_request_late_transform"
  zone_id = "64cbd9e9124a7e8a20bddc8365bfd3b9"

rules {
    action = "rewrite"
    action_parameters {
      headers {
        expression = "ip.src"
        name       = "X-test-Forwarded-For"
        operation  = "set"
      }
    }
      expression  = "true"
      description = "replication"
      enabled     = true
    }
  }
```


### Link to debug output

https://gist.github.com/chris-martinelli/dfb181e0d04b12e084673ca6384332e2

### Panic output

[debug.log](https://github.com/cloudflare/terraform-provider-cloudflare/files/11658542/debug.log)


### Expected output

Following the steps outlined in https://registry.terraform.io/providers/cloudflare/cloudflare/latest/docs/guides/version-4-upgrade#cloudflare_ruleset, the expectation is to be able to import this resource after upgrading from v3.35.0 to v4.7.1.

### Actual output

```
cloudflare_ruleset.transform_rules: Importing from ID "cd72d32b51904560b408a7d720df15a3"...
╷
│ Error: Plugin did not respond
│
│ The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).ImportResourceState call. The plugin logs may contain more details.
╵


Stack trace from the terraform-provider-cloudflare_v4.7.1 plugin:

panic: runtime error: index out of range [1] with length 1
```

### Steps to reproduce

1) Having a cloudflare_ruleset rule in working order using provider v3.35.0

```
cloudflare_ruleset.transform_rules: Creation complete after 1s [id=cd72d32b51904560b408a7d720df15a3]
```

2) Remove the resource

```
terraform state rm cloudflare_ruleset.transform_rules
```

3) Upgrade to v4.7.1
(Updated to version = "~> 4.0" in .tf file)

```
terraform init -upgrade
```

4) Import previously removed resource

```
terraform import cloudflare_ruleset.transform_rules cd72d32b51904560b408a7d720df15a3
```

### Additional factoids

_No response_

### References

https://github.com/cloudflare/terraform-provider-cloudflare/issues/2501

--- a/.changelog/2503.txt
+++ b/.changelog/2503.txt
@@ -1,0 +1,3 @@
+```release-note:bug
+resource/cloudflare_ruleset: handle `Import` operations where the required values are missing for providing a nicer error message
+```

--- a/internal/framework/service/rulesets/resource.go
+++ b/internal/framework/service/rulesets/resource.go
@@ -267,15 +267,15 @@ func (r *RulesetResource) Delete(ctx context.Context, req resource.DeleteRequest
 
 func (r *RulesetResource) ImportState(ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse) {
 	idParts := strings.Split(req.ID, "/")
-	resourceLevel, resourceIdentifier, rulesetID := idParts[0], idParts[1], idParts[2]
 
-	if len(idParts) != 3 || resourceLevel == "" || resourceIdentifier == "" || rulesetID == "" {
+	if len(idParts) != 3 || idParts[0] == "" || idParts[1] == "" || idParts[2] == "" {
 		resp.Diagnostics.AddError(
 			"invalid import identifier",
 			fmt.Sprintf("expected import identifier to be resourceLevel/resourceIdentifier/rulesetID. got: %q", req.ID),
 		)
 		return
 	}
+	resourceLevel, resourceIdentifier, rulesetID := idParts[0], idParts[1], idParts[2]
 
 	resp.Diagnostics.Append(resp.State.SetAttribute(ctx, path.Root("id"), rulesetID)...)
 	if resourceLevel == "zone" {

--- a/internal/framework/service/rulesets/resource_test.go
+++ b/internal/framework/service/rulesets/resource_test.go
@@ -2197,6 +2197,27 @@ func TestAccCloudflareRuleset_CacheSettingsDefinedQueryStringIncludeKeys(t *test
 	})
 }
 
+func TestAccCloudflareRuleset_ImportHandlesMissingValues(t *testing.T) {
+	rnd := utils.GenerateRandomResourceName()
+	zoneID := os.Getenv("CLOUDFLARE_ZONE_ID")
+	zoneName := os.Getenv("CLOUDFLARE_DOMAIN")
+	name := "cloudflare_ruleset." + rnd
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.TestAccPreCheck(t) },
+		ProtoV6ProviderFactories: acctest.TestAccProtoV6ProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config:        testAccCheckCloudflareRulesetTransformationRuleResponseHeaders(rnd, "broken", zoneID, zoneName),
+				ExpectError:   regexp.MustCompile(`invalid import identifier`),
+				ImportState:   true,
+				ImportStateId: rnd,
+				ResourceName:  name,
+			},
+		},
+	})
+}
+
 func testAccCheckCloudflareRulesetMagicTransitSingle(rnd, name, accountID string) string {
 	return fmt.Sprintf(`
   resource "cloudflare_ruleset" "%[1]s" {

